lang/syn/src/idl/parse/file.rs

```diff
@@ -270,7 +270,7 @@ fn parse_account_derives(ctx: &CrateContext) -> HashMap<String, AccountsStruct>
         .filter_map(|i_strct| {
             for attr in &i_strct.attrs {
                 if attr.path.is_ident("derive") && attr.tokens.to_string().contains(DERIVE_NAME) {
-                    let strct = accounts::parse(i_strct).expect("Code not parseable");
+                    let strct = accounts::parse(i_strct).unwrap();
```

Comment:
How does the output look like now? I feel like this could make it even worse because at least we were able to search for `Code not parseable` in the codebase and find where it panics. Maybe better to combine both?

Reply:
Thank you for reviewing @acheroncrypto. 

For now, the output look like this. The error message shows correctly when I run both command `anchor build`, `anchor test`. 

![Screenshot from 2023-11-16 08-48-43](https://github.com/coral-xyz/anchor/assets/62386689/436f261f-b0e2-45ca-a7f1-bf1f6d4b8f95)

So should I leave this issue?

Reply:
This change doesn't affect that error message though, it only affects IDL generation and you'll get the same compiler error with or without this change.

Running `anchor idl parse --file programs/program/src/lib.rs` outputs:

```
thread 'main' panicked at lang/syn/src/idl/parse/file.rs:273:58:
Code not parseable: Error("expected `,`")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

And with your changes:

```
thread 'main' panicked at lang/syn/src/idl/parse/file.rs:273:58:
called `Result::unwrap()` on an `Err` value: Error("expected `,`")
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

so there is no improvement with this change.

> For now, the output look like this. The error message shows correctly when I run both command `anchor build`, `anchor test`.

#683 is a very old issue so it's likely that this got fixed somewhere along the line because both `anchor build` and `anchor test` seem to show the same error message.

Reply:
Ok. Thank you.